contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+
+        return Decimal.ratio(targetBalance, totalBalance);
+    }
+
+    function amountFromTarget(address pcvDeposit) external view virtual override returns(int256) {
+        uint totalBalance = getTotalBalance();
+
+        uint pcvDepositBalance = IPCVDeposit(pcvDeposit).balance();
+        uint pcvDepositWeight = pcvDepositInfos[address(pcvDeposit)].weight;
+
+        uint idealDepositBalance = pcvDepositWeight * totalBalance / totalWeight;
+        
+        return int(idealDepositBalance) - int(pcvDepositBalance);
+    }
+
+    function pcvDeposits() external view virtual override returns(address[] memory deposits, uint256[] memory weights) {
+        address[] memory _deposits = new address[](pcvDepositAddresses.length());
+        uint256[] memory _weights = new uint256[](pcvDepositAddresses.length());
+
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            _deposits[i] = pcvDepositAddresses.at(i);
+            _weights[i] = pcvDepositInfos[pcvDepositAddresses.at(i)].weight;
+        }
+
+        return (_deposits, _weights);
+    }
+
+    function getTotalBalance() public view virtual override returns (uint256) {
+        uint totalBalance = 0;
+
+        for (uint i=0; i<pcvDepositAddresses.length(); i++) {
+            totalBalance += IPCVDeposit(pcvDepositAddresses.at(i)).balance();
+        }
+
+        // Let's not forget to get this balance
+        totalBalance += IERC20(token).balanceOf(address(this));
+
+        return totalBalance;
+    }
+
+    // ---------- Internal Functions ----------- //
+
+    function _rebalanceSingle(address pcvDeposit) internal {
+        if (!pcvDepositAddresses.contains(pcvDeposit)) {
+            revert("Deposit does not exist.");
+        }
+
+        uint totalBalance = getTotalBalance();
+        uint128 pcvDepositWeight = pcvDepositInfos[pcvDeposit].weight;
+        uint idealDepositBalance = pcvDepositWeight * totalBalance / totalWeight;
+        uint pcvDepositBalance = IPCVDeposit(pcvDeposit).balance();
+
+        if (pcvDepositBalance > idealDepositBalance) {
+            // PCV deposit balance is too high. Withdraw from it into the aggregator.
+            uint overage = pcvDepositBalance - idealDepositBalance;
+            IPCVDeposit(pcvDeposit).withdraw(address(this), overage);
+        } else if (pcvDepositBalance < idealDepositBalance) {
+            // PCV deposit balance is too low. Pull from the aggregator balance if we can.
+            uint defecit = idealDepositBalance - pcvDepositBalance;
+            if (IERC20(token).balanceOf(address(this)) >= defecit) {
+                IERC20(token).safeTransfer(pcvDeposit, defecit);
+                IPCVDeposit(pcvDeposit).deposit();
+            } else {
+                // Emit event so that we know to do a full rebalance
+            }
+        } else {
+            // PCV deposit balance is exactly where it needs to be. Don't touch it.
+        }
+    }
+
+
+    function _rebalance() internal {
```

Comment:
We should try to improve the algorithm here. Shouldn't do more than 2 external calls to each PCV deposit, one for balance, and one for withdraw/deposit

Reply:
This is currently the case, save for the ones we deposit into, where we call deposit() after transferring.

Reply:
Caching implemented.